File: db/event_role.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_role.
*/

class Event_Role extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_role";
    $this->order = "";
    $this->domain = "valuelist";
    $this->field['event_role_id']['type'] = 'SERIAL';
    $this->field['event_role_id']['not_null'] = true;
    $this->field['tag']['type'] = 'VARCHAR';
    $this->field['tag']['length'] = 32;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['event_role_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_role.rb
module Momomoto
  class Event_role < Base
    def initialize
      super
      @domain = 'valuelist'
      @fields = {
        :event_role_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :tag => Datatype::Varchar.new( {:not_null=>true, :length=>32} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
